File: gatekeeper/internal/modules/backoffice-proxy/dto/question_dto.go
package dto

import "github.com/dev-oleksandrv/easy-pobyt/gatekeeper/internal/model"

type QuestionContentOutputDto struct {
	Text        string  `json:"text"`
	Explanation string  `json:"explanation"`
	ImageURL    *string `json:"image_url"`
}

type QuestionAnswerOutputDto struct {
	Text      string `json:"text"`
	IsCorrect bool   `json:"is_correct"`
}

type QuestionContentGroupOutputDto struct {
	Lang    string                    `json:"lang"`
	Content QuestionContentOutputDto  `json:"content"`
	Answers []QuestionAnswerOutputDto `json:"answers"`
}

type QuestionOutputDto struct {
	ID        string                          `json:"id"`
	Type      string                          `json:"type"`
	Contents  []QuestionContentGroupOutputDto `json:"contents"`
	CreatedAt string                          `json:"created_at"`
	UpdatedAt string                          `json:"updated_at"`
}

type QuestionContentInputDto struct {
	Text        string  `json:"text"`
	Explanation string  `json:"explanation"`
	ImageURL    *string `json:"image_url"`
}

type QuestionAnswerInputDto struct {
	Text      string `json:"text"`
	IsCorrect bool   `json:"is_correct"`
}

type QuestionContentGroupInputDto struct {
	Lang    model.LanguageCode       `json:"lang"`
	Content QuestionContentInputDto  `json:"content"`
	Answers []QuestionAnswerInputDto `json:"answers"`
}

type QuestionInputDto struct {
	Type     model.QuestionType             `json:"type"`
	Contents []QuestionContentGroupInputDto `json:"contents"`
}

type QuestionBatchInputDto struct {
	Questions []QuestionInputDto `json:"questions"`
}

File: backoffice-client/dto/question-dto.ts
import { QuestionType } from "@/types/question-types";
import { LanguageCode } from "@/types/common-types";

export interface QuestionAnswerDto {
  text: string;
  is_correct: boolean;
}

export interface QuestionContentDto {
  text: string;
  explanation: string;
  image_url: string;
}

export interface QuestionContentGroupDto {
  lang: LanguageCode;
  content: QuestionContentDto;
  answers: QuestionAnswerDto[];
}

export interface QuestionDto {
  id: string;
  type: QuestionType;
  contents: QuestionContentGroupDto[];
}

export interface QuestionBatchCreateResponseDto {
  data: QuestionDto[];
}

export interface QuestionGetListResponseDto {
  data: QuestionDto[];
}
